### PR TITLE
Remove noautovalidity from pNexts in vk.xml

### PR DESCRIPTION
**Pull request by krOoze**
ex #1284

diff:
diff1
diff2
diff3
diff4

### CODE REVIEW

File: xml/vk.xml [hunk 1]
```diff
@@ -3163,7 +3163,7 @@ typedef void <name>CAMetalLayer</name>;
         </type>
         <type category="struct" name="VkDebugUtilsMessengerCallbackDataEXT">
             <member values="VK_STRUCTURE_TYPE_DEBUG_UTILS_MESSENGER_CALLBACK_DATA_EXT"><type>VkStructureType</type> <name>sType</name></member>
-            <member optional="true">const <type>void</type>*                                                        <name>pNext</name></member>
+            <member>const <type>void</type>*                                                                        <name>pNext</name></member>
```
Comment (krOoze) [inline xml/vk.xml:3166]: Plus remove this `optional`, which likely has no purpose.

**Comment by jeffbolznv** [comment]
Many of these look wrong. The purpose of `noautovalidity` is to make it so only the base structure in a chain gets the auto-generated VU, and I think this matches how the layers implement it. The styleguide says (about `structextends`):

```
  * attr:structextends ...
    This should usually only list the top-level structure that is
    extended, for all possible chained structures - the chained structures
    themselves should have their `pNext` tagged with attr:noautovalidity.
```

Some of them, like the renderpass2 stuff, look like correct changes, but I think all the Features/Properties structs should be reverted.

**Comment by krOoze** [comment]
In that case, it is actually mising in many places. E.g. `VkPhysicalDeviceVariablePointersFeatures`. If layers relied on this, it would be broken.

Perphaps ping me what is the chosen way forward before I instead start adding `noautovalidity`ies at bulk into the XML.

**Comment by oddhack** [comment]
If everyone's now agreed that structures with the structextends attribute should also have noautovalidity on pNext, which will suppress the "pNext must be NULL" VUs, I'll take care of that. It's easier for us to handle those sorts of bulk XML edits internally. If there's somewhere else that you think noautovalidity does or does not belong, or something else like the 'optional' you refer to above, please bring that out as a separate issue.

**Comment by oddhack** [comment]
Hm. Adding or removing noautovalidity from pNext actually makes no difference at all in the generated VU for extending structures. @tobski would probably have to explain the guts of the validity generator but this makes sense to me. Actually the most useful thing to do might probably be to remove the comment from the style guide about setting noautovalidity on pNext in extending structures, since it's irrelevant.

**Comment by krOoze** [comment]
@oddhack Yes, the generator does not care. It cares only if it sees `structextends`, and so it knows it is extension struct. And as such the `novalidity` does nothing. Though that logic is slightly broken, as a `struct` could be both extension and top-level, as is the case of e.g. `VkDebugUtilsMessengerCreateInfoEXT`.

https://github.com/KhronosGroup/Vulkan-Docs/blob/68bedacc6014941d5c04d168ef62df82031e0ddc/scripts/validitygenerator.py#L1012-L1017

**Pull request closed without merging**